File: uncovering-history-spring/src/main/java/com/uncovering_history/api/model/InterestPoint.java
package com.uncovering_history.api.model;

import java.time.LocalDateTime;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;

@Data
@Document(collection = "interest_point")
public class InterestPoint {

    @Id
    private String objectId;

    @Indexed
    private String name;
    private String tipo;

    private String endereco;
    private String descripcion;
    private String bairro;
    private String contato;
    private String personalizados;
    private List<String> fotos;
    private List<String> elementos;
    private String idAuthor;
    private Double lat;
    private Double lon;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;


}

File: historiador_app/lib/db/isar/interest_point.dart
import 'package:isar/isar.dart';
import 'elemento_pi.dart';

part 'interest_point.g.dart';

@collection
class InterestPoint {
  Id id = Isar.autoIncrement;

  String? objectId;
  late String name;
  late String tipo;
  late String endereco;
  late String descripcion;
  late String bairro;
  String? contato;
  String? personalizados;
  List<String> fotos = [];
  List<String> elementos = [];
  late String idAuthor;
  
  late double lat;
  late double lon;

  List<Id> elementosIsar = [];
  
  DateTime createdAt = DateTime.now();
  DateTime updatedAt = DateTime.now();

}